app/controllers/results_controller.rb

```diff
@@ -59,8 +59,10 @@ def edit
     @first_file = @files.first
     @extra_marks_points = @result.extra_marks.points
     @extra_marks_percentage = @result.extra_marks.percentage
-    @marks_map = Hash.new
-    @old_marks_map = Hash.new
+    @flexible_marks_map = Hash.new
```

Comment:
Also, if it seems like I am passing a lot of variables, I could use the fact that instance variables can be accessed directly anywhere down the road, but it is good practice to pass variables to partials.